botocore/credentials.py

```diff
@@ -82,6 +84,8 @@ def create_credential_resolver(session):
             creds_filename=credential_file,
             profile_name=profile_name
         ),
+        ProcessProvider(profile_name=profile_name,
```

Comment:
Can the `ProcessProvider` take precedence over the `SharedCredentialProvider` or would this cause issues?

I can submit a PR with the change if there's no obvious reasons not to do it. It'll allow me to re-use a saml provider CLI tool more efficiently.

Reply:
Existing providers always take precedence over new providers (except for EC2 metadata, which is always last). This is to make sure that people's config files continue to work the same as they had before if they don't change them.

Regardless, all you will need to do to make sure you're using this provider is to make a new profile where it's the only one set and use that profile explicitly.

Reply:
Guess where it feels a bit inconsistent to me is `source_profile`/`role_arn` win over hardcoded credentials:
```
[profile]
source_profile = rd
role_arn       = arn:blah:blah
# These will be ignored and it will use assumeRole provider
aws_access_key_id = ...
aws_secret_access_key = ...
aws_session_token = ...
```

But when using the `credential_process` it's the other way around, the hardcoded credentials win over the config option.

Reply:
The ability to have hard-coded credentials in a profile along with static credentials was never intended behavior. We won't break that (because people rely on it now), but we also don't want to continue doing that for other providers.